airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -36,25 +34,26 @@ class HttpStream(Stream, ABC):
 
     # TODO: remove legacy HttpAuthenticator authenticator references
     def __init__(self, authenticator: Union[AuthBase, HttpAuthenticator] = None):
-        self._session = requests.Session()
+        if self.use_cache:
+            self._session = self.request_cache()
+        else:
+            self._session = requests.Session()
 
         self._authenticator: HttpAuthenticator = NoAuth()
         if isinstance(authenticator, AuthBase):
             self._session.auth = authenticator
         elif authenticator:
             self._authenticator = authenticator
 
-        if self.use_cache:
-            self.cache_file = self.request_cache()
-            # we need this attr to get metadata about cassettes, such as record play count, all records played, etc.
-            self.cassete = None
-
     @property
     def cache_filename(self):
         """
         Override if needed. Return the name of cache file
         """
-        return f"{self.name}.yml"
+        number = get_instance_number(self)
```

Comment:
Why can't multiple users of this HTTP class use the same sqlite file?  Won't /something/ in the request be different (cookie, URL, params) be different if needed... otherwise isn't it valuable to share the same cache file?

Reply:
same q as evan